### PR TITLE
Add react-toastify to change notification logic.

### PR DESCRIPTION
**Pull request by tuovinensanttu**
## Description

Installed `react-toastify` and wrapper the old `NotificationComponent` with it. 

## Context

This change allows us to get rid of the logic where `NotificationComponent` was imported to almost every component.

## How Has This Been Tested?
Tested this manually.

## Manual Testing Instructions for Reviewers

One way to test this is to use renew membership functionality.

### CODE REVIEW

File: src/domain/auth/components/login/Login.tsx [hunk 1]
```diff
@@ -42,6 +42,10 @@ function Login(props: Props) {
 
   const isAuthenticated = useSelector(isAuthenticatedSelector);
 
+  if (Boolean(props.auth.error)) {
+    toastNotification({ onClose: () => props.resetError() });
+  }
```
Comment (tuovinensanttu) [inline src/domain/auth/components/login/Login.tsx:47]: This solution seems to be working, only one notification is shown at the time. I'm still thinking if I should wrap this with `useEffect` to be sure.

File: CHANGELOG.md [hunk 2]
```diff
@@ -36,6 +36,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Changed
 
 - Use GitHub actions instead of Travis
+- centralize notification with react-toastify.
```
Comment (Jokinen) [inline CHANGELOG.md:39]: Minor, no capitalisation

File: src/common/components/notification/toastNotification.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+export type NotificationOptions = {
+  labelText?: string;
+  notificationMessage?: string;
+  type?: 'info' | 'success' | 'alert' | 'error';
+  onClose?: () => void;
+};
+
+const toastNotification = (notificationOptions: NotificationOptions) => {
```
Comment (Jokinen) [inline src/common/components/notification/toastNotification.tsx:20]: This is stored under `components`. Is it really a component?

File: src/domain/app/App.tsx [hunk 4]
```diff
@@ -19,7 +21,7 @@ import authConstants from '../../domain/auth/constants/authConstants';
 import authenticate from '../../domain/auth/authenticate';
 import AppMeta from './AppMeta';
 import AppRoutes from './AppRoutes';
-
+import styles from './app.module.css';
```
Comment (Jokinen) [inline src/domain/app/App.tsx:24]: Missing whitespace.

File: src/domain/app/AppYouthProfileRoute.tsx [hunk 5]
```diff
@@ -29,12 +29,9 @@ function AppYouthProfileRoute({
   ...rest
 }: Props) {
   const { t } = useTranslation();
-  const [showNotification, setShowNotification] = useState<boolean>(false);
 
   const { data, loading } = useQuery<HasYouthProfile>(HAS_YOUTH_PROFILE, {
-    onError: () => {
-      setShowNotification(true);
-    },
+    onError: () => toastNotification({}),
```
Comment (Jokinen) [inline src/domain/app/AppYouthProfileRoute.tsx:34]: Using an empty object is discouraged.

File: src/domain/auth/components/login/Login.tsx (hunk 1 above)
Comment (Jokinen) [reply src/domain/auth/components/login/Login.tsx:47]: Well, you are correct. The render function should not have side effects unless wrapped in a `useEffect`. But I think that an even better route could be to set the toast in the authentication routine itself, in which case you would not need to use a `useEffect` either. Now the firing of the notification doesn't need to be tied to this component anymore.

**Review by Jokinen — COMMENTED** [review]
Good changes! Added a few comments.

File: src/domain/app/AppYouthProfileRoute.tsx (hunk 5 above)
Comment (tuovinensanttu) [reply src/domain/app/AppYouthProfileRoute.tsx]: Fixed now. Silly mistake on my part to, forgot to make whole `notificationOptions` optional.

File: src/common/components/notification/toastNotification.tsx (hunk 3 above)
Comment (tuovinensanttu) [reply src/common/components/notification/toastNotification.tsx]: True! Moved it under helpers instead.

File: CHANGELOG.md (hunk 2 above)
Comment (tuovinensanttu) [reply CHANGELOG.md]: Fixed!

**Comment by tuovinensanttu** [comment]
> Well, you are correct. The render function should not have side effects unless wrapped in a useEffect. But I think that an even better route could be to set the toast in the authentication routine itself, in which case you would not need to use a useEffect either. Now the firing of the notification doesn't need to be tied to this component anymore.

Seems that I was so used to notification being a component that I didn't even think about this. Removed error handling from `Login` component and moved it to `authenticate` and `logout` methods.

**Review by Jokinen — APPROVED** [review]

**Pull request merged**